### DISCUSSION TITLE
How to quote front-matter variables in main body

### DISCUSSION THREAD

**Discussion by Nemo1166** [vuepress/core#1098] [Q&A] [answered]
As I want to create .md templates for re-use, I planned to put some properties into YAML front-matter then wrote a .md file like this:

```markdown
---
......
property1: xxx
propertygroup: [xxx, yyy, zzz]
propertydict:
  sub1: xxx
  sub2: yyy
---
<!-- template.md -->
......
some text {{ Page.property1 }} and {{ Page.propertygroup[0] }}, {{ propertydict.sub1 }}
```

Unfortunately, it didn't work. So how to quote the variables in front-matter correctly? Or should I take other extra actions to make it?

Forgive my lack of knowledge about vue.

Additional info about runtime

- vue: 3.2.36
- vuepress: 2.0.0-beta.48
- vuepress-theme-hope: 2.0.0-beta.84

**Comment by Mister-Hope** [comment, marked as answer]
$page.frontmatter.xxx

**Reply by Nemo1166** [reply]
It works! thx a lot ♥

**Reply by Mister-Hope** [reply]
BTW, you may check if there is variable `$frontmatter`

**Reply by Nemo1166** [reply]
> BTW, you may check if there is variable `$frontmatter`

`$frontmatter.xxx` totally works.

In addition, here is a further problem: how can I set `vuepress-plugin-md-enhance` rendering after mustache stuff parsed? cuz I wanna write the template like this:

```markdown
::: tabs#id

@tab {{ $frontmatter.branch[0] }}

xxx

@tab {{ $frontmatter.branch[1] }}

yyy

:::
```
and here is what it rendered

![image](https://user-images.githubusercontent.com/38162382/191903393-7e48e856-3d79-4123-bf41-0a38b0a1f6a7.png)

**Reply by Mister-Hope** [reply]
content after tab is extracted by markdown-it plugin, yet it doesnot exisit by raw text. Note, you can only use that syntax in text node.

Just another example:

```md
[a]({{ b }}/README.md)
```

Doesn't work because it's :


```html
<a href="{{ b }}/index.html">a</a>
```

while `[{{ a }}](b/README.md)` work, because the `{{ a }}` is a text node (not attr or other thing)
